backup/moodle2/backup_applaunch_activity_task.class.php

```diff
@@ -0,0 +1,76 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Define backup task.
+ *
+ * @package    mod_applaunch
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2021 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+
+require_once($CFG->dirroot . '/mod/applaunch/backup/moodle2/backup_applaunch_stepslib.php');
+
+class backup_applaunch_activity_task extends backup_activity_task {
+
+    /**
+     * Define particular settings this activity can have.
+     */
+    protected function define_my_settings() {
+        // No particular settings for this activity.
+    }
+
+    /**
+     * Define particular steps this activity can have.
+     */
+    protected function define_my_steps() {
+        $this->add_step(new backup_applaunch_activity_structure_step('applaunch_structure', 'applaunch.xml'));
```

Comment:
Should the pathing for this be inside an activities folder or similar? Also is the .xml necessary?

Reply:
I'm not 100% sure what you mean by this. This puts the step data in the main applaunch.xml inside the course module dir in the backup zip. It's the same pattern as the other activities.

Reply:
I haven't done a lot of backup/restore code so don't have a full understanding of it.

Reply:
Example backup folder - 
[backup-totara-course-5-2_1-20210608-1529.zip](https://github.com/catalyst/moodle-mod_applaunch/files/6635177/backup-totara-course-5-2_1-20210608-1529.zip)